### ISSUE TITLE
How to obtain the accuracy of the model?

### ISSUE THREAD

**Issue by Robin2091** [zzh8829/yolov3-tf2#116]
In the train.py file I changed the parameters of the model to this:

model.compile(optimizer=optimizer, loss=loss,
                      run_eagerly= False, metrics = ['accuracy'])
When I get the accuracy reading outputted, it is in two parts: yolo_0_accuracy and yolo_1_accuracy. Same thing with the validation accuracy. How do I obtain the total accuracy of the model?

Thank you.

**Comment by nivsmall** [comment]
I'm guessing you are using YoloTiny?

If so it kind of makes sense (i played around in eager excecution so my evaluations were a little different)

the "tiny" model detects objects in two different scales (in comparison to 3 scales in normal YoloV3). When not in training (hence in test/detect) it uses non max suppression (nms) to choose the right detection out of the proposed ones (again out of two different scales) but during training it doesn't.

that basically means that during training the model detects twice for each image (or for each grid cell in image to be precise). If i understand correctly that is why you are getting metrics for two separate detection instead of one

I guess you could average the accuracy if it is important for you to get a single metric but mind you the it will probably give you a lower accuracy then in actual test/detection (since nms doesn't choose detections randomly)
Would be great for someone else to verify this answer and the details in it...

best of luck!

**Comment by Robin2091** [comment]
@nivsmall Thank you for your answer. I want to clarify if mAP is the same thing as accuracy in this model. If not, how can I find the mAP for my model.

Once again, thank you for your reply.

**Comment by zzh8829** [comment]
i don't believe the code does mAP calculation yet, merging this with https://github.com/zzh8829/yolov3-tf2/issues/125
